modules/mapbox/src/deck-utils.ts

```diff
@@ -60,7 +60,7 @@ export function getDeckInstance({
     });
     // If using the WebGLContext created by deck (React use case), we use deck's viewState to drive the map.
     // Otherwise (pure JS use case), we use the map's viewState to drive deck.
-    map.on('move', () => onMapMove(deckInstance, map));
+    map.on('move', () => deckInstance.isInitialized && onMapMove(deckInstance, map));
```

Comment:
Alternatively, we could try to call `map.off`, but it's stateful requiring the original function be passed in.

Reply:
Is there an event deck fires when it is finalized? We could utilize that here since `overlay.finalize()` calls `deck.finalize()`, and we have access to the deck instance here.

Reply:
I think the most common way to trigger this is by removing MapboxOverlay? In which case you can just call something from the overlay's `onRemove`.

If you want to address the more generic case, I guess you can make the callback deregister itself after the first time it's called with deck finalized.

Reply:
Yeah, `onRemove` in the overlay is the natural place to call something. https://github.com/visgl/deck.gl/pull/7649/commits/2f7fde03324e55aaf0c644daf270894fcb42490e shows what that could look like in the overlay, except `MapboxLayer` needs a one-time `map.on('move')` applied so I can't simply delete it in `getDeckInstance`.

Reply:
I'm leaning towards preferring the generic solution where the callback deregisters itself... I see we don't have a `deck.isFinalized` and that many of the signals may either mean yet-to-be-initialized or finalized (https://github.com/visgl/deck.gl/blob/ef8ea132f694aa1d47b1f2e3a262c388534eb6ba/modules/core/src/lib/deck.ts#L484-L485). Do you know what we can use to detect that deck is finalized?